This patch is to resolve the API performance issue by removing the cache generation which is run on each product save,
we will instead run this process manually.
--- vendor/magento/module-catalog/Model/Product.php	2017-02-16 10:59:18.000000000 +0000
+++ vendor/magento/module-catalog/Model/Product.php	2017-07-31 09:58:17.000000000 +0100
@@ -923,13 +923,6 @@
         $this->_getResource()->addCommitCallback([$this, 'reindex']);
         $this->reloadPriceInfo();
 
-        // Resize images for catalog product and save results to image cache
-        /** @var Product\Image\Cache $imageCache */
-        if (!$this->_appState->isAreaCodeEmulated()) {
-            $imageCache = $this->imageCacheFactory->create();
-            $imageCache->generate($this);
-        }
-
         return $result;
     }
 
